js/functions.js

```diff
@@ -2211,3 +2218,134 @@ function getEventDescription(isAerobatics, locationName, minutes) {
+    //var timeCount = 0;
+    function notifyUserIfNear(currentLocation, aircraft) {
+        if (userLoc) {
+            currentLocation = {lon: currentLocation.lng, lat: currentLocation.lat};
+            if(haversineDistance(userLoc,currentLocation) < 2000)
+            {
+                notifiedNearUser = true;
+                //check that its not the same popup that has been closed
+                /*if(name === "")
+                {
+                    console.log("in timer");
+                    var x = setInterval( function () { timeCount++;}, 1000);
+                } */
+                //check that its not the same popup that has been closed - if its not the same aircraft or its has been more then 30 sec
+                //if(timeCount >30 || name !== aircraft.name) {
+                    if(document.getElementById("myModal").style.display === "none" && document.getElementById("gottoVoiceMessagePopup").style.display === "none")
+                    {
+                        //timeCount = 0;
+                        name = aircraft.name;
+                        document.getElementById("gottoVoiceMessagePopup").style.display = "block";
+                        document.getElementById("aircraftName").innerHTML = " מטוס קרב " + name;
+                        document.getElementById("aircraftTime").innerHTML = "יעבור מעלייך 30 שניות";
+                        document.getElementById("youHaveVoicemessage").innerHTML = "יש לך הודעה קולית מהטייס!";
+                        document.getElementById("voiceMessageImg").src = "icons/voiceMessage/dictation_glyph.png";
+                        
+                        var engName="";
+                        var audioSrc = "";
+                        for (var i=0; i<aircraftsName.length; i++){
+                            if(aircraftsName[i].hebName == name){
+                                engName = aircraftsName[i].engName;
+                                audioSrc = aircraftsName [i].audioSrc;
+                                break;
+                            }
+                        }
+                        if(audioSrc !== undefined && audioSrc !== ""){
+                            document.getElementById("audioSRC").src=audioSrc;
+                        }
+                        else{
+                            document.getElementById("audioSRC").src="";
+                        }
+
+                        if(engName !== undefined && engName !== ""){
+                            document.getElementById("aircraftImg").src="icons/aircrafts/"+engName+".png";
+                        }
+                        else{
+                            document.getElementById("aircraftImg").src="";
+                        }
+                    }
+                //}
+            }
+
+            else if(haversineDistance(userLoc,currentLocation) > 2000)
+            {
+                if(document.getElementById("myModal").style.display !== "block")
```

Comment:
try to find cleaner way to check if it is visible... perhaps use a variable.

Reply:
I will check @AsafKaravani